tests/test_state.py

```diff
@@ -1,657 +1,660 @@
-        child_state: A child state.
-        grandchild_state: A grandchild state.
-    """
-    # Events should bubble down to the substate.
-    assert child_state.value == ""
-    assert child_state.count == 23
-    event = Event(
-        token="t", name="child_state.change_both", payload={"value": "hi", "count": 12}
-    )
-    update = await test_state.process(event)
-    assert child_state.value == "HI"
-    assert child_state.count == 24
-    assert update.delta == {
-        "test_state.child_state": {"value": "HI", "count": 24},
-        "test_state": {"sum": 3.14, "upper": ""},
-    }
-    test_state.clean()
-
-    # Test with the granchild state.
-    assert grandchild_state.value2 == ""
-    event = Event(
-        token="t",
-        name="child_state.grandchild_state.set_value2",
-        payload={"value": "new"},
-    )
-    update = await test_state.process(event)
-    assert grandchild_state.value2 == "new"
-    assert update.delta == {
-        "test_state.child_state.grandchild_state": {"value2": "new"},
-        "test_state": {"sum": 3.14, "upper": ""},
-    }
-
-
-def test_format_event_handler():
-    """Test formatting an event handler."""
-    assert (
-        utils.format_event_handler(TestState.do_something) == "test_state.do_something"  # type: ignore
-    )
-    assert (
-        utils.format_event_handler(ChildState.change_both)  # type: ignore
-        == "test_state.child_state.change_both"
-    )
-    assert (
-        utils.format_event_handler(GrandchildState.do_nothing)  # type: ignore
-        == "test_state.child_state.grandchild_state.do_nothing"
-    )
-
-
-def test_get_token(test_state):
-    assert test_state.get_token() == ""
-
-    token = "b181904c-3953-4a79-dc18-ae9518c22f05"
-    test_state.router_data = {RouteVar.CLIENT_TOKEN: token}
-
-    assert test_state.get_token() == token
-
-
-def test_get_current_page(test_state):
-
-    assert test_state.get_current_page() == ""
-
-    route = "mypage/subpage"
-    test_state.router_data = {RouteVar.PATH: route}
-
-    assert test_state.get_current_page() == route
-
-
-def test_get_query_params(test_state):
-    assert test_state.get_query_params() == {}
-
-    params = {"p1": "a", "p2": "b"}
-    test_state.router_data = {RouteVar.QUERY: params}
-
-    assert test_state.get_query_params() == params
-
-
-def test_add_var(test_state):
-    test_state.add_var("dynamic_int", int, 42)
-    assert test_state.dynamic_int == 42
-
-    test_state.add_var("dynamic_list", List[int], [5, 10])
-    assert test_state.dynamic_list == [5, 10]
-    assert getattr(test_state, "dynamic_list") == [5, 10]
-
-    # how to test that one?
-    # test_state.dynamic_list.append(15)
-    # assert test_state.dynamic_list == [5, 10, 15]
-
-    test_state.add_var("dynamic_dict", Dict[str, int], {"k1": 5, "k2": 10})
-    assert test_state.dynamic_dict == {"k1": 5, "k2": 10}
-    assert getattr(test_state, "dynamic_dict") == {"k1": 5, "k2": 10}
+from typing import Dict, List

```

Comment:
Looks good, but why is the diff so large on this file? (I think there may be some extra carriage returns or some weird thing like that) - can we resolve this

Reply:
Line breaks in Windows and Unix-like OS? When I switch from Windows to Linux a few months ago, similar thing happend.

Reply:
I think this was caused line breaks too.
The diff is only few line at TestState class.

```py
class TestState(State):
    """A test state."""

    # Set this class as not test one   # <- these 2 lines
    __test__ = False                   # <-
```

I think I should revert them before the commit in order to delete line breaks, then commit again.
What do you think?

Reply:
I think that should work try that, or rebase on the current main branch and commit again